typings/styled-components.d.ts

```diff
@@ -207,9 +96,12 @@ export function injectGlobal(strings: TemplateStringsArray, ...interpolations: S
 
 export const ThemeProvider: ThemeProviderComponent<any>;
 
+export class StyleSheetManager extends React.Component<{ sheet: ServerStyleSheet }, any> { }
+
 export class ServerStyleSheet {
-  collectStyles(children: ReactElement<any>): ReactElement<any>
-  getStyleTags(): string
+  collectStyles(children: ReactElement<any>): StyleSheetManager
```

Comment:
By looking [the docs](https://www.styled-components.com/docs/advanced#server-side-rendering) I figure `collectStyles` doesn't really return a `StyleSheetManager` instance, it returns _a react element_. So maybe:
```ts
collectStyles(tree: React.ReactNode): ReactElement<{ sheet: ServerStyleSheet; }>;
```

And `{ sheet: ServerStyleSheet; }` can be defined in its own interface since it's reused.